client/js/components/GameView/MoveChoiceOverlay.vue

```diff
@@ -70,181 +70,167 @@ export default {
     },
   },
   computed: {
-    /**
-     * Returns list of objects representing the available moves,
-     * based on the selected card
-     */
-    moveChoices() {
-      if (!this.selectedCard) return [];
-      let res = [];
-      let cardRank;
-      if (this.resolvingSeven) {
-        if (!this.cardSelectedFromDeck) return [];
-        cardRank = this.cardSelectedFromDeck.rank;
-      } else {
-        if (!this.selectedCard) return [];
-        cardRank = this.selectedCard.rank;
-      }
-      // Re-usable move-objects
-      const allMovesAreDisabled = !this.isPlayersTurn || this.frozenId === this.selectedCard.id;
-      const allMovesDisabledExplanation = !this.isPlayersTurn
-        ? "It's not your turn"
-        : 'This card is frozen';
-
-      let disabledText;
-      if (allMovesAreDisabled) {
-        disabledText = allMovesDisabledExplanation;
+    // Determines if any moves are available
+    allMovesAreDisabled() {
+      return !this.isPlayersTurn || this.frozenId === this.selectedCard.id;
+    },
+    // Determines which disabled text to display
+    disabledText() {
+      if (this.allMovesAreDisabled) {
+        return !this.isPlayersTurn ? "It's not your turn" : 'This card is frozen';
       }
-
-      // Points
-      const pointsDescription = `Gain ${cardRank} point` + (cardRank === 1 ? '' : 's');
-      const pointsMove = {
+      return '';
+    },
+    pointsMove() {
+      const pointsDescription =
+        `Gain ${this.selectedCard.rank} point` + (this.selectedCard.rank === 1 ? '' : 's');
+      return {
         displayName: 'Points',
         eventName: 'points',
         moveDescription: pointsDescription,
-        disabled: allMovesAreDisabled,
-        disabledExplanation: disabledText,
+        disabled: this.allMovesAreDisabled,
+        disabledExplanation: this.disabledText,
       };
-      // Scuttling
+    },
+    scuttleMove() {
       const scuttleDisabled = !this.isPlayersTurn || !this.hasValidScuttleTarget;
       let scuttleDisabledExplanation = 'You can only scuttle smaller point cards';
       if (this.$store.getters.opponent.points.length === 0) {
         scuttleDisabledExplanation = 'Your opponent has no point cards to scuttle';
       }
-      if (allMovesAreDisabled) {
-        scuttleDisabledExplanation = allMovesDisabledExplanation;
+      if (this.allMovesAreDisabled) {
+        scuttleDisabledExplanation = this.disabledText;
       }
-      const scuttleMove = {
+      return {
         displayName: 'Scuttle',
         eventName: 'scuttle',
         moveDescription: 'Scrap a lower point card',
-        disabled: scuttleDisabled || allMovesAreDisabled,
+        disabled: scuttleDisabled || this.allMovesAreDisabled,
```

Comment:
With the above change this line can just use the existing variable without ||
```suggestion
        disabled: scuttleDisabled,
```

Reply:
This change also causes a few test failures, one being `Disables playing a frozen number card until the following turn` from oneOffs. The frozen disabled text is missing from the scuttle move choice with this change in place. Reverting it causes the test to pass.

Reply:
I think it's because of my typo above. See the revised suggestion